### ISSUE TITLE
VPN clients isolated.

### ISSUE THREAD

**Issue by m0lmk** [hwdsl2/setup-ipsec-vpn#215]
I know the default config is set so VPN clients are not isolated but that's not what I am seeing.

I can ping each client from the server but I can not ping another client from a client.

I'm just running the default iptables rules so can't work out why this is not working.

Any ideas on how to fix this?

**Comment by hwdsl2** [comment]
@m0lmk The default configuration by the VPN setup scripts does allow communication among VPN clients. Regarding your issue: First, many operating systems (e.g. Android) disallow pings by default. In addition, when communicating among the VPN clients, make sure that you use their assigned internal VPN IPs (not their public IPs). For example, for IPsec/L2TP the VPN clients are assigned IPs from range 192.168.42.10-192.168.42.250, and for IPsec/XAuth they are assigned IPs from range 192.168.43.10-192.168.43.250. To see which internal IP is assigned, check the VPN connection details.

**Comment by m0lmk** [comment]
This is what I am doing. I have assigned static IPs to the VPN clients.

Server: 192.168.42.1
Client 1: 192.168.42.10
Client 2: 192.168.42.11

On server:
`ping 192.168.42.10`
OK
`ping 192.168.42.11`
OK

So the clients are responding to pings from the server.

From client 1:
`ping 192.168.42.1`
OK
`ping 192.168.42.11`
No reply

From client 2:
`ping 192.168.42.1`
OK
`ping 192.168.42.10`
No reply

I can ping the clients from the server which proves they are connected and responding to pings. I can also ssh into the clients from the server using their VPN IP so I know that the ssh port is open and working.

I can't ping or ssh into either client from the other client.

**Comment by hwdsl2** [comment]
@m0lmk Check your IPTables FORWARD rules:
```
iptables -nvL FORWARD
```

If the packet counters in front of rule # 4, i. e. `-A FORWARD -i ppp+ -o ppp+ -s 192.168.42.0/24 -d 192.168.42.0/24 -j ACCEPT`, is not zero, it means that traffic between IPsec/L2TP VPN clients were allowed by the VPN server, and your issue is probably caused by something else.

**Comment by m0lmk** [comment]
Here's what I have:

`$ iptables -nvL FORWARD`
`Chain FORWARD (policy ACCEPT 0 packets, 0 bytes)`
`pkts bytes target     prot opt in     out     source               destination`
`482  143K DOCKER-USER  all  --  *      *       0.0.0.0/0            0.0.0.0/0`
`482  143K DOCKER-ISOLATION  all  --  *      *       0.0.0.0/0            0.0.0.0/0`
`0     0 ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED`
`0     0 DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0`
`0     0 ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0`
`0     0 ACCEPT     all  --  docker0 docker0  0.0.0.0/0            0.0.0.0/0`
`0     0 ACCEPT     all  --  *      br-bc6ba59050a9  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED`
`0     0 DOCKER     all  --  *      br-bc6ba59050a9  0.0.0.0/0            0.0.0.0/0  `         
`0     0 ACCEPT     all  --  br-bc6ba59050a9 !br-bc6ba59050a9  0.0.0.0/0            0.0.0.0/0   `        
`0     0 ACCEPT     all  --  br-bc6ba59050a9 br-bc6ba59050a9  0.0.0.0/0            0.0.0.0/0 `          
`0     0 ACCEPT     all  --  br-3c99f603f42d br-3c99f603f42d  0.0.0.0/0            0.0.0.0/0  `         
`2    80 DROP       all  --  *      *       0.0.0.0/0            0.0.0.0/0            ctstate INVALID`
`190 79313 ACCEPT     all  --  eth0   ppp+    0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED`
`258 61262 ACCEPT     all  --  ppp+   eth0    0.0.0.0/0            0.0.0.0/0 `          
`32  2619 ACCEPT     all  --  ppp+   ppp+    192.168.42.0/24      192.168.42.0/24`     
`0     0 ACCEPT     all  --  eth0   *       0.0.0.0/0            192.168.43.0/24      ctstate RELATED,ESTABLISHED`
`0     0 ACCEPT     all  --  *      eth0    192.168.43.0/24      0.0.0.0/0`           
`0     0 ACCEPT     all  --  *      *       192.168.42.0/24      192.168.42.0/24`     
`0     0 DROP       all  --  *      *       0.0.0.0/0            0.0.0.0/0`


So it look like it's working but it's not.

I played around with adding some new rules and eventually found that adding these rules makes it work:

`iptables -t nat -A POSTROUTING -s 192.168.42.0/24 -d 192.168.42.0/24 -j MASQUERADE`

I've added `-A POSTROUTING -s 192.168.42.0/24 -d 192.168.42.0/24 -j MASQUERADE` to `/etc/iptables.rules` and it all seems to be working after a reboot.

Odd!
